pkg/controller/backup/backup_controller.go

```diff
@@ -0,0 +1,196 @@
+}
+
+func Add(mgr manager.Manager) error {
+	return zero.CreateController(ControllerName, &reconcileBackup{mgr.GetClient()}, mgr)
+}
+
+func (r *reconcileBackup) ResourceInstance(key client.ObjectKey, ctrl *zero.Controller) (zero.Resource, error) {
+	instance := &v2.Backup{}
+	if err := ctrl.Get(ctx, key, instance); err != nil {
+		return nil, err
+	}
+
+	instance.Spec.ApplyDefaults()
+	return &backupResource{
+		instance: instance,
+		client:   r.Client,
+		scheme:   ctrl.Scheme,
+	}, nil
+}
+
+func (r *reconcileBackup) Type() runtime.Object {
+	return &v2.Backup{}
+}
+
+func (r *backupResource) AsMeta() metav1.Object {
+	return r.instance
+}
+
+func (r *backupResource) Cluster() string {
+	return r.instance.Spec.Cluster
+}
+
+func (r *backupResource) Phase() zero.Phase {
+	return zero.Phase(string(r.instance.Status.Phase))
+}
+
+func (r *backupResource) UpdatePhase(phase zero.Phase) error {
+	instance := r.instance
+	instance.Status.Phase = v2.BackupPhase(string(phase))
+	return r.updateStatus()
+}
+
+func (r *backupResource) Init() (*zero.Spec, error) {
+	err := r.getOrCreatePvc()
+	if err != nil {
+		return nil, err
+	}
+
+	r.instance.Status.PVC = fmt.Sprintf("pvc/%s", r.instance.Name)
+	if err = r.updateStatus(); err != nil {
+		return nil, err
+	}
+
+	return &zero.Spec{
+		Volume: zero.VolumeSpec{
+			MountPath: DataMountPath,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: r.instance.Name,
+				},
+			},
+		},
+		Container: r.instance.Spec.Container,
+		PodLabels: PodLabels(r.instance.Name, r.instance.Spec.Cluster),
+	}, nil
+}
+
+func (r *backupResource) getOrCreatePvc() error {
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := r.client.Get(ctx, client.ObjectKey{
+		Name:      r.instance.Name,
+		Namespace: r.instance.Namespace,
+	}, pvc)
+
+	// If the pvc already exists simply return
+	if err == nil {
+		return nil
+	}
+
+	if !errors.IsNotFound(err) {
+		return err
+	}
+
+	volumeSpec := r.instance.Spec.Volume
+	var storage resource.Quantity
+	if volumeSpec.Storage == nil {
+		// TODO calculate based upon number of Pods in cluster
+		// ISPN- Utilise backup size estimate
+		storage = constants.DefaultPVSize
+	} else {
+		storage, err = resource.ParseQuantity(*volumeSpec.Storage)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO add labels
+	pvc = &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.instance.Name,
+			Namespace: r.instance.Namespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: storage,
+				},
+			},
+			StorageClassName: volumeSpec.StorageClassName,
+		},
+	}
+	controllerutil.SetControllerReference(r.instance, pvc, r.scheme)
+	pvc.OwnerReferences[0].BlockOwnerDeletion = pointer.BoolPtr(false)
```

Comment:
@ryanemerson I've validated this option on OCP 4.5 and looks like not need anymore, as was a bug. But I don't know what is this functionality target OCP version.

Reply:
`master` is going to target 4.5, so no need for this workaround :slightly_smiling_face: